pkg/kubelet/kubelet.go

```diff
@@ -1405,10 +1414,7 @@ func (kl *Kubelet) GenerateRunContainerOptions(pod *api.Pod, container *api.Cont
 		return nil, err
 	}
 	opts.Hostname = hostname
-	vol, ok := kl.volumeManager.GetVolumes(pod.UID)
-	if !ok {
-		return nil, fmt.Errorf("impossible: cannot find the mounted volumes for pod %q", format.Pod(pod))
-	}
+	vol := kl.volumeManager.GetVolumesForPodAndAppendSupplementalGroups(pod)
```

Comment:
In a follow-up, as we discussed in video chat: 

1.  Add a field to `RunContainerOptions` for extra supplemental groups
2.  Change this method to `GetVolumesAndVolumeSupplementalGroupsForPod(*api.Pod) ([]volume, []groups, error)`
3.  Populate new field from (1) with supplemental groups
4.  Change SecurityContextProvider interface to accept an extra list of supplemental groups to `ModifyHostConfig`
5.  TODO for `rkt`

...that only solves the pod mutation problem.  Still left to solve is validating that the user can use the group.